### PR TITLE
Fixes and minor cleanups to pod (anti)affinity predicate

### PR DESCRIPTION
**Pull request by wojtek-t**
@davidopp @bsalamat 

Cherrypick of #45098

### CODE REVIEW

File: plugin/pkg/scheduler/algorithm/predicates/predicates.go [hunk 1]
```diff
@@ -1074,14 +1074,13 @@ func getMatchingAntiAffinityTerms(pod *v1.Pod, nodeInfoMap map[string]*scheduler
 				continue
 			}
 			for _, term := range getPodAntiAffinityTerms(affinity.PodAntiAffinity) {
-				namespaces := priorityutil.GetNamespacesFromPodAffinityTerm(pod, &term)
+				namespaces := priorityutil.GetNamespacesFromPodAffinityTerm(existingPod, &term)
```
Comment (mwielgus) [inline plugin/pkg/scheduler/algorithm/predicates/predicates.go]: Why do we use a different pod here?

Comment (wojtek-t) [inline plugin/pkg/scheduler/algorithm/predicates/predicates.go]: this was a bug fix
A separate PR with test for it was added in https://github.com/kubernetes/kubernetes/pull/45152

**Comment by mwielgus** [comment]
@ethernetdan @enisoc 
We need this PR to be merged before the next 1.6 release is cut.

**Comment by wojtek-t** [comment]
@enisoc
This PR fixes a bug in pod anti-affinity in scheduler (it was also very helpful to facilitate debugging of cluster autoscaler, as we were silently dropping errors). As the next cut is planned for the beginning of the next week, you are on vacation and running the e2e tests can take a day I will allow myself to apply the cherrypick-approved label for this PR so you can proceed with the release as soon as you are back.

**Comment by wojtek-t** [comment]
@k8s-bot etcd3 test this
@k8s-bot kubemark test this

**Pull request merged**